shipping_zone.go

```diff
@@ -0,0 +1,96 @@
+package goshopify
+
+import (
+	"github.com/shopspring/decimal"
+)
+
+// ShippingZoneService is an interface for interfacing with the shipping zones endpoint
+// of the Shopify API.
+// See: https://help.shopify.com/api/reference/store-properties/shippingzone
+type ShippingZoneService interface {
+	List() ([]ShippingZone, error)
+}
+
+// ShippingZoneServiceOp handles communication with the shipping zone related methods
+// of the Shopify API.
+type ShippingZoneServiceOp struct {
+	client *Client
+}
+
+// ShippingZone represents a Shopify shipping zone
+type ShippingZone struct {
+	ID                           int64                         `json:"id,omitempty"`
+	Name                         string                        `json:"name,omitempty"`
+	ProfileID                    string                        `json:"profile_id,omitempty"`
+	LocationGroupID              string                        `json:"location_group_id,omitempty"`
+	AdminGraphqlAPIID            string                        `json:"admin_graphql_api_id,omitempty"`
+	Countries                    []ShippingCountry             `json:"countries,omitempty"`
+	WeightBasedShippingRates     []WeightBasedShippingRate     `json:"weight_based_shipping_rates,omitempty"`
+	PriceBasedShippingRates      []PriceBasedShippingRate      `json:"price_based_shipping_rates,omitempty"`
+	CarrierShippingRateProviders []CarrierShippingRateProvider `json:"carrier_shipping_rate_providers,omitempty"`
+}
+
+// ShippingCountry represents a Shopify shipping country
+type ShippingCountry struct {
+	ID             int64              `json:"id,omitempty"`
+	ShippingZoneID int64              `json:"shipping_zone_id,omitempty"`
+	Name           string             `json:"name,omitempty"`
+	Tax            *decimal.Decimal   `json:"tax,omitempty"`
+	Code           string             `json:"code,omitempty"`
+	TaxName        string             `json:"tax_name,omitempty"`
+	Provinces      []ShippingProvince `json:"provinces,omitempty"`
+}
+
+// ShippingProvince represents a Shopify shipping province
+type ShippingProvince struct {
+	ID             int64            `json:"id,omitempty"`
```

Comment:
Should we include the province code fidel that is mentioned in the [docs](https://shopify.dev/docs/admin-api/rest/reference/store-properties/shippingzone?api[version]=2019-10)?